sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/rel/BeamValuesRel.java

```diff
@@ -82,7 +82,7 @@ public BeamValuesRel(
 
       Schema schema = CalciteUtils.toSchema(getRowType());
       List<Row> rows = tuples.stream().map(tuple -> tupleToRow(schema, tuple)).collect(toList());
-      return pinput.getPipeline().begin().apply(Create.of(rows)).setRowSchema(schema);
+      return pinput.getPipeline().begin().apply(Create.of(rows).withRowSchema(schema));
```

Comment:
Ah. schema should be set on Create.of() but not PCollection<Row>, as the latter is too late.